### PR TITLE
Preliminary tag mapping

### PR DESCRIPTION
**Pull request by andrewpollock**
This is dreadfully inefficient in the long run, but great for proof-of-concept purposes.

I'm not sure why `go.sum` has gone bonkers?

### CODE REVIEW

File: vulnfeeds/cpp/main.go [hunk 1]
```diff
@@ -73,21 +75,65 @@ func InScopeGitRepo(repoURL string) bool {
...
+		Logger.Infof("Cloning %s", repo)
+		r, err := git.Clone(memory.NewStorage(), nil, &git.CloneOptions{
+			URL: repo,
+		})
+		if err != nil {
+			Logger.Warnf("Failed clone %s: %v", repo, err)
+			continue
+		}
+		for _, av := range versions.AffectedVersions {
+			if av.Fixed == "" {
```
Comment (oliverchang) [inline vulnfeeds/cpp/main.go:93]: Can you add a comment on what this case means and why we skip?

File: vulnfeeds/cpp/main.go [hunk 2]
```diff
@@ -73,21 +75,65 @@ func InScopeGitRepo(repoURL string) bool {
...
+				continue
+			}
+			// Need to come up with a way to decide when to prefix
+			ref, err := r.Tag("v" + av.Fixed)
```
Comment (oliverchang) [inline vulnfeeds/cpp/main.go:97]: Let's add a TODO to support some more sophisticated/fuzzy tag -> version matching.

File: vulnfeeds/cpp/main.go [hunk 3]
```diff
@@ -73,21 +75,65 @@ func InScopeGitRepo(repoURL string) bool {
...
+			if err != nil {
+				Logger.Warnf("Failed to find a tag for %q in %s: %v", av.Fixed, repo, err)
```
Comment (oliverchang) [inline vulnfeeds/cpp/main.go:99]: I wonder if these need to be surfaced as "notes" for potential human triagers.

**Review by oliverchang — COMMENTED** [review]
nice! 

IIUC `go.sum` is not meant to be a lockfile -- it contains additional state necessary for https://research.swtch.com/vgo-mvs.

**Review by oliverchang — APPROVED** [review]
LGTM with minor nits.

File: vulnfeeds/cpp/main.go (hunk 3 above)
Comment (andrewpollock) [reply vulnfeeds/cpp/main.go:99]: Yes. At the moment I'm not generating per-record notes at all, but I see the value in doing this, and will implement this in the not too distant future.

File: vulnfeeds/cpp/main.go (hunk 1 above)
Comment (andrewpollock) [reply vulnfeeds/cpp/main.go:118]: Addressed in 7ed688a7d9e25ab364426d06556083b9ffa0229f

File: vulnfeeds/cpp/main.go (hunk 2 above)
Comment (andrewpollock) [reply vulnfeeds/cpp/main.go:122]: Addressed in 7ed688a7d9e25ab364426d06556083b9ffa0229f

**Pull request merged**